worker/src/crypto.rs

```diff
@@ -0,0 +1,65 @@
+use js_sys::{ArrayBuffer, Uint8Array};
+use wasm_bindgen::prelude::*;
+use wasm_bindgen_futures::JsFuture;
+
+/// A Rust-friendly wrapper around the non-standard [crypto.DigestStream](https://developers.cloudflare.com/workers/runtime-apis/web-crypto/#constructors) API
+///
+/// Example usage:
+/// ```rust
+/// let digest_stream = DigestStream::new(DigestStreamAlgorithm::Sha256);
+///
+/// // create a ReadableStream from a string
+/// let mut req_init = web_sys::RequestInit::new();
+/// req_init.method("POST");
+/// req_init.body(Some(&JsValue::from_str("foo")));
+/// let req = web_sys::Request::new_with_str_and_init("http://internal", &req_init).unwrap();
+/// let body = req.body().unwrap();
+///
+/// // just kick the promise off to the background, we'll await the digest itself
+/// // since this is piped to a JS readable stream, call .raw() to get the underlying JS object
+/// let _ = body.pipe_to(digest_stream.raw());
+///
+/// let bytes:Vec<u8> = digest_stream.digest().await.unwrap().to_vec();
+/// ```
+pub struct DigestStream {
+    inner: worker_sys::DigestStream,
+}
+
+impl DigestStream {
+    #[allow(dead_code)]
+    pub fn new(algo: DigestStreamAlgorithm) -> Self {
+        Self {
+            inner: worker_sys::DigestStream::new(algo.as_str()),
+        }
+    }
+
+    #[allow(dead_code)]
+    pub async fn digest(&self) -> Result<Uint8Array, crate::Error> {
+        let buffer: ArrayBuffer = JsFuture::from(self.inner.digest()).await?.unchecked_into();
```

Comment:
You may need to do this to avoid `Sync` / `Send` issues. 
```suggestion
        let fut = SendFuture(JsFuture::from(self.inner.digest()));
        let buffer: ArrayBuffer = fut.await?.unchecked_into();
```

Reply:
interesting... I haven't used `SendFuture` and don't quite understand it since wasm can't actually send futures between threads... I suppose it's for making the futures work with "multithreaded" (but not really) frameworks?

anyway - no objection from me, committing

Reply:
Yep, exactly, we need to tell Rust that its safe in this context.